File: app/systems/Message.lua
local ____lualib = require("lualib_bundle")
local __TS__SourceMapTraceBack = ____lualib.__TS__SourceMapTraceBack
__TS__SourceMapTraceBack(debug.getinfo(1).short_src, {["5"] = 5,["6"] = 10,["7"] = 12,["8"] = 5,["9"] = 15,["10"] = 16,["11"] = 15});
local ____exports = {}
function ____exports.postMessage(self, receiver, messageId, message)
    local receiverUrl = type(receiver) == "string" and msg.url(receiver) or receiver
    msg.post(receiverUrl, messageId, message)
end
function ____exports.messageUrl(self, ref)
    return msg.url(ref)
end
return ____exports

File: src/modules/systems/Message.ts
import { Message } from '../types/Message';
import { MessageId } from '../types/MessageId';
import { Ref } from '../types/Url';

export function postMessage<ID extends MessageId>(
	receiver: Ref | url,
	messageId: ID,
	message: Message<ID>,
) {
	const receiverUrl =
		typeof receiver === 'string' ? msg.url(receiver) : receiver;
	msg.post(receiverUrl, messageId, message);
}

export function messageUrl(ref: Ref) {
	return msg.url(ref);
}
